apps/wizard/pages/fasttrack/app.py

```diff
@@ -158,6 +164,36 @@
         ),
     )
 
+# These need to be defined outside of the form to be able to make the `is_public` checkbox dependent
+# on the `existing_google_sheet` value
+if import_method == UPDATE_GSHEET:
+    if placeholder_for_existing_google_sheet is None:
+        raise ValueError("placeholder_for_existing_google_sheet is None. This was not expected.")
+    else:
+        with placeholder_for_existing_google_sheet:
+            existing_google_sheet = st.selectbox(
+                label="Existing Google Sheets",
+                options=options,  # type: ignore
+                format_func=lambda x: x["label"],
+                help="Selected sheet will be used if you don't specify Google Sheets URL.",
+                key="dataset_uri",
+                on_change=reset_states,
```

Comment:
This is what I have added to ensure that code after line 201 is not executed whenever the user changes this state after having clicked on "Submit" earlier.